### ISSUE TITLE
Support declaring queues as auto-delete

### ISSUE THREAD

**Issue by axelo** [insurello/Insurello.RabbitMqClient#9]
### Problem

Would be nice with RPC queues to be marked as auto-delete, so when it's get out of style (no longer used), the queue will be removed for us by the broker.

### Solution

https://www.rabbitmq.com/queues.html#temporary-queues

**Comment by Mousaka** [comment]
Yes! It also might be even better to use [Queue TTL](https://www.rabbitmq.com/ttl.html#queue-ttl) which seems to work like auto-delete but you can also have a time limit. This post recommends it and it seems to make sense https://stackoverflow.com/a/58181544 . We currently have a lot of idle temp queues lying around in prod.
